actionpack/test/controller/routing_test.rb

```diff
@@ -361,6 +361,22 @@ def test_route_with_regexp_and_captures_for_controller
     assert_raise(ActionController::RoutingError) { rs.recognize_path("/admin/products") }
   end
 
+  def test_route_with_rack_environment
+    rs.draw do
+      post   "/people"     => "people#create"
+      get    "/people/:id" => "people#show"
+      put    "/people/:id" => "people#update"
+      patch  "/people/:id" => "people#update"
+      delete "/people/:id" => "people#destroy"
+    end
+
+    assert_equal({controller: "people", action: "create"}, rs.recognize_path("/people", Rack::MockRequest.env_for("/people", method: :post)))
```

Comment:
Looks like this method's behavior was moved to `recognize` and the PR should be closed.

https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/journey/router.rb#L58